### PR TITLE
FIX error responses section in NGSIv2 spec

### PR DESCRIPTION
**Pull request by fgalan**
Section "Error responses" has been re-written, based on https://github.com/telefonicaid/fiware-orion/issues/1286 explaining with more detail how error are generated in NGSIv2.

I suggest not ussing diff at github, which is pretty confusing, but to read the new version directly.

### CODE REVIEW

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 1]
```diff
@@ -396,30 +396,30 @@ If present, the error payload is a JSON object including the following fields:
...
-* RequestEntityTooLarge (`413`). Payload is too large. The specific limit to consider a payload too large depends
-                                 on the particular implementation.
-* UnsupportedMediaType  (`415`). Request content type is not supported.
-* InvalidModification   (`422`). Some piece of information is missing in payload.
-* NotSupportedQuery     (`422`). The implementation does not support the query issued.
-* NoResourcesAvailable  (`413`). There are no server resources to fulfill the client request. 
-
-New error codes may be defined in new iterations of this specification.
+
+All NGSIv2 server implementations must use the following HTTP response codes and `error` texts
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: HTTP status codes

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 2]
```diff
@@ -396,30 +396,30 @@ If present, the error payload is a JSON object including the following fields:
...
+described in this section. However, the particular text used for `description` field is
+an implementation specific aspect.
+
+NGSIv2 `error` reporting is as follows:
+
++ If the incoming JSON payload cannot be parsed then `ParseError` (`400`) is return.
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: returned

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 3]
```diff
@@ -396,30 +396,30 @@ If present, the error payload is a JSON object including the following fields:
...
++ Request error which only depends on the request itself (i.e. it does not depend on the NGSIv2 server status),
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Errors which are only caused by request content

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 4]
```diff
@@ -396,30 +396,30 @@ If present, the error payload is a JSON object including the following fields:
...
+  either in the URL parameters or in the payload, results in `BadRequest`(`400`).
+  + Exception: incoming JSON payload errors, which have another `error` message (see previous bullet).
++ Attemp to exceed spatial index limit results in `NoResourceAvailable` (`413`). See "Geospatial properties of entities"
+  section for details.
++ Ambiguety due to the request may refer to several resources, e.g. attempt to update an entity providing only its ID
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: check spelling

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 1 above)
Comment (fgalan) [reply doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in adac43d7713c1ebee422cea5033fbb3f1c651502

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 2 above)
Comment (fgalan) [reply doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in adac43d7713c1ebee422cea5033fbb3f1c651502

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 3 above)
Comment (fgalan) [reply doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in adac43d7713c1ebee422cea5033fbb3f1c651502

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 4 above)
Comment (fgalan) [reply doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in adac43d7713c1ebee422cea5033fbb3f1c651502

**Pull request merged**